File: temp.py
import time, os
import datetime
import rrdtool
import socket 
import json
#import couchdb
#couch = couchdb.Server('http://192.168.8.1:5984/')
db = couch['sensors']
beauty_ip = "10.70.60.1"
beauty_port = 2003
path = "/sys/bus/w1/devices/"
tempSensor = { 'beagle' : '10-0008029674ee', 'out' : '10-000802964c0d', 'cylinder' : '10-000802961f0d', 'stove' : '10-00080296978d', 'canal' : '28-000004ee99a8' }


def timestamp():
    timestamp = json.dumps(datetime.datetime.now().isoformat()).split('"')[1]
    return timestamp

def getTemp(sensor,address):
    if os.path.isfile(path+address+"/w1_slave"):
        try:
            raw = open(path+address+"/w1_slave", "r").read()
            if 'YES' in raw:
                current_temp = float(raw.split("t=")[-1])/1000
                return current_temp
            else:
                return 'crc_error'
        except IOError:
            print "%s %s is offline" % (sensor,address)
def printTemp():
    pass


sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM) 
while True:
    now = str(time.time())
    for k,v in tempSensor.iteritems():
        temp=str(getTemp(k,v))
        if 'crc_error' not in temp:
            graph  = ".".join(['boat.temp',k])
            packet = " ".join([graph, temp, now])
            print "%s  : %s C" %(k, str(temp))
            sock.sendto(packet, (beauty_ip,beauty_port))
    time.sleep(60)
    print "----------"

    

File: lib/boattr/sensors/temp.rb
module Boattr
  class Sensors
    class Temp
      attr_reader :name, :address
      def initialize(name, address)
        @name     = name
        @address  = address
      end

      def read
        begin
          @file = File.open("/sys/bus/w1/devices/#{address}/w1_slave", 'r')
        rescue
          return
        end
        return unless @file.readline.include?('YES') # Is CRC valid in the first line?
        @temp = @file.readline.split[-1].split('=')[-1].to_i / 1000.0
        { 'name' => @name, 'type' => 'temp', 'address' => address,  'value' => @temp.round(2) }
      end
    end
  end
end
